src/crate/theme/vendor/rtd_compat/extension.py

```diff
@@ -1,255 +0,0 @@
-        # However, we are replacing them with the new Addons.
-        # I wouldn't include them in the first version of the extension.
-        # We could hardcode them if we want, tho.
-        #
-        # 'MEDIA_URL': "{{ settings.MEDIA_URL }}",
-        # 'STATIC_URL': "{{ settings.STATIC_URL }}",
-        # 'proxied_static_path': "{{ proxied_static_path }}",
-
-        'PRODUCTION_DOMAIN': production_domain,
-        'versions': versions,
-        "downloads": downloads,
-        "subprojects": subprojects,
-
-        'slug': project_slug,
-        'rtd_language': os.environ.get("READTHEDOCS_LANGUAGE"),
-        'canonical_url': os.environ.get("READTHEDOCS_CANONICAL_URL"),
-
-        # NOTE: these seem to not be used.
-        # 'name': u'{{ project.name }}',
-        # 'analytics_code': '{{ project.analytics_code }}',
-        # 'single_version': {{ project.is_single_version }},
-        # 'programming_language': u'{{ project.programming_language }}',
-
-        'conf_py_path': conf_py_path,
-        # Used only for "readthedocs-sphinx-ext" which we are not installing anymore.
-        # 'api_host': '{{ api_host }}',
-        # 'proxied_api_host': '{{ project.proxied_api_host }}',
-
-        'github_user': github_user,
-        'github_repo': github_repo,
-        'github_version': os.environ.get("READTHEDOCS_GIT_IDENTIFIER"),
-        'display_github': github_user is not None,
-        'bitbucket_user': bitbucket_user,
-        'bitbucket_repo': bitbucket_repo,
-        'bitbucket_version': os.environ.get("READTHEDOCS_GIT_IDENTIFIER"),
-        'display_bitbucket': bitbucket_user is not None,
-        'gitlab_user': gitlab_user,
-        'gitlab_repo': gitlab_repo,
-        'gitlab_version': os.environ.get("READTHEDOCS_GIT_IDENTIFIER"),
-        'display_gitlab': gitlab_user is not None,
-        'READTHEDOCS': True,
-        'using_theme': (config.html_theme == "default"),
-        'new_theme': (config.html_theme == "sphinx_rtd_theme"),
-        'source_suffix': ".rst",
-        'ad_free': False,
-        'docsearch_disabled': False,
-
-        # We don't support Google analytics anymore.
-        # See https://github.com/readthedocs/readthedocs.org/issues/9530
-        'user_analytics_code': "",
-        'global_analytics_code': None,
-
-        'commit': os.environ.get("READTHEDOCS_GIT_COMMIT_HASH")[:8],
-    }
-
-    # For sphinx >=1.8 we can use html_baseurl to set the canonical URL.
-    # https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_baseurl
-    if version_info >= (1, 8):
-        if not hasattr(config, 'html_baseurl'):
-            config.html_baseurl = context['canonical_url']
-        context['canonical_url'] = None
-
-
-    if hasattr(config, 'html_context'):
-        for key in context:
-            if key not in config.html_context:
-                config.html_context[key] = context[key]
-    else:
-        config.html_context = context
```

Comment:
As you can see [here](https://github.com/crate/crate-docs-theme/blob/0.38.3/src/crate/theme/rtd/conf/__init__.py#L292-L297) and [there](https://github.com/crate/crate-docs-theme/blob/0.38.3/src/crate/theme/rtd/conf/__init__.py#L312-L314), we applied trial-and-error techniques in the past to make things work without knowing too much about relevant details.

Maybe this is the key element we have been missing before: First, populate the template `context`. Then, propagate all ingredients into the `html_context` 1:1, in a generic way. I think we may want to carry this gem over to our `crate/theme/rtd/conf/__init__.py` as well.